### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by roaksoax**

### CODE REVIEW

File: docs/index.asciidoc [hunk 1]
```diff
@@ -171,6 +172,16 @@ Authenticate using Elasticsearch API key. Note that this option also requires en
...
 
+[id="plugins-{type}s-{plugin}-proxy"]
+===== `proxy`
+
+* Value type is <<uri,uri>>
+* There is no default value for this setting.
+
+Set the address of a forward HTTP proxy.
+An empty string is treated as if proxy was not set, this is useful when using
+environment variables e.g. `proxy => '${LS_PROXY:}'`.
```
Comment (karenzone) [inline docs/index.asciidoc:183]:
```suggestion
An empty string is treated as if proxy was not set, and is useful when using
environment variables e.g. `proxy => '${LS_PROXY:}'`.
```

**Comment by kares** [comment]
@yaauie raised concerns about relying on internals when this was done on the ES input (after the output was merged), 
so we did it differently: https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114/files ... not sure it matters much esp. if the integration plugin is down on us sooner than later, in which case we could unify easily.

**Comment by roaksoax** [comment]
@kares can you clarify your comment? Either way, my changes are very similar to yours with the exception that it is using 'uri' instead of 'uri_or_empty', and validates 'uri' instead.

**Comment by kares** [comment]
@roaksoax see the concerns raised at the linked PR: https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114#pullrequestreview-346986372
ES output was done the same way - using `config_init` which is internal API compared to using `validate_value`

**Comment by roaksoax** [comment]
@yaauie thoughts on the above?

**Comment by yaauie** [comment]
@roaksoax I would echo the above-linked review I made of the ES Input, for the exact same reasons. Overriding `config_init` has side effects (e.g., changing the plugin's record of explicit as-given arguments `@original_params`), and overriding the validation step only more appropriately matches the desired scope.

**Pull request merged**